src/components/index.js

```diff
@@ -9,11 +9,11 @@ import {Ul} from './common';
 import defaultDecorators from './Decorators';
 import TreeNode from './TreeNode';
 
-const TreeBeard = ({animations, decorators, data, onToggle, style}) => (
+const TreeBeard = ({animations, decorators, data, separateToggleEvent, onToggle, onClickHeader, style}) => (
     <Ul style={{...defaultTheme.tree.base, ...style.tree.base}}>
         {castArray(data).map(node => (
             <TreeNode
-                {...{decorators, node, onToggle, animations}}
+                {...{decorators, node, separateToggleEvent, onToggle, onClickHeader, animations}}
```

Comment:
please create unit test when you press `onClickHeader` or when you have  `onClickHeader `